### PR TITLE
feat: update dolphin settings

### PR DESCRIPTION
**Pull request by cnkeats**
Pushes settings from Launcher to Dolphin when Dolphin is launched. Launcher settings are preferred and used to overwrite Dolphin settings when reasonable (time to launch dolphin is not noticeably affected).
- Replay root directory
- Use monthly subfolders

Users that don't use the Launcher will be able to change their settings as normal. However, if everyone is eventually required to use the Launcher, the ability to change settings via Dolphin should either be removed or synced with the Launcher.

image

### CODE REVIEW

File: src/dolphin/manager.ts [hunk 1]
```diff
@@ -46,6 +46,8 @@ export class DolphinManager extends EventEmitter {
       throw new Error("Netplay dolphin is already open!");
     }
 
+    void updateDolphinSettings();
```
Comment (vinceau) [inline src/dolphin/manager.ts:49]: Why are we `void`ing these promises?

```suggestion
    await updateDolphinSettings();
```

File: src/dolphin/manager.ts [hunk 2]
```diff
@@ -62,6 +64,9 @@ export class DolphinManager extends EventEmitter {
 
   public async configureDolphin(launchType: DolphinLaunchType) {
     log.debug(`configuring ${launchType} dolphin...`);
+
+    void updateDolphinSettings();
```
Comment (vinceau) [inline src/dolphin/manager.ts:68]:
```suggestion
    await updateDolphinSettings();
```

File: src/renderer/components/PathInput.tsx [hunk 3]
```diff
@@ -30,9 +32,13 @@ export const PathInput = React.forwardRef<HTMLInputElement, PathInputProps>((pro
         <CustomInput inputRef={ref} disabled={true} value={value} placeholder={placeholder} />
         {endAdornment}
       </InputContainer>
-      <Button color="secondary" variant="contained" onClick={onClick} disabled={disabled}>
-        Select
-      </Button>
+      <Tooltip title={(disabled && tooltipText) ?? ""}>
```
Comment (vinceau) [inline src/renderer/components/PathInput.tsx:35]: Why is this necessary to check for disabled? Why can't we just pass the tooltip text straight in?

```suggestion
      <Tooltip title={tooltipText}>
```

File: src/settings/settingsManager.ts [hunk 4]
```diff
@@ -45,12 +46,26 @@ export class SettingsManager {
     }
   }
 
+  public getRootSlpPath(): string {
+    return this.appSettings.settings?.rootSlpPath as string;
```
Comment (vinceau) [inline src/settings/settingsManager.ts:50]: If you're needing to cast this, you're doing something wrong. You should actually handle the `.settings` case being undefined and not use optional chaining.

File: src/settings/settingsManager.ts [hunk 5]
```diff
@@ -45,12 +46,26 @@ export class SettingsManager {
...
+  }
+
+  public getUseMonthlySubfolders(): boolean {
+    return this.appSettings.settings?.useMonthlySubfolders as boolean;
```
Comment (vinceau) [inline src/settings/settingsManager.ts:54]: Again you should be handling the undefined case instead of using optional chaining and type casting.

**Review by vinceau — CHANGES_REQUESTED** [review]

File: src/renderer/components/PathInput.tsx (hunk 3 above)
Comment (NikhilNarayana) [reply src/renderer/components/PathInput.tsx:35]: right that tooltip is kinda specific for the disabled case. I wanted to move it outside the component but that didn't work cause it would be on the entire component. We could just show the tooltip at all times I guess.

Comment (vinceau) [reply src/renderer/components/PathInput.tsx:35]: Couldn't the component passing in the `tooltipText` to `PathInput` pass in undefined if not `disabled` then?

Comment (NikhilNarayana) [reply src/renderer/components/PathInput.tsx]: can't be undefined, but yeah it can pass in an empty string

File: src/renderer/components/PathInput.tsx [hunk 6]
```diff
@@ -30,9 +32,13 @@ export const PathInput = React.forwardRef<HTMLInputElement, PathInputProps>((pro
         <CustomInput inputRef={ref} disabled={true} value={value} placeholder={placeholder} />
         {endAdornment}
       </InputContainer>
-      <Button color="secondary" variant="contained" onClick={onClick} disabled={disabled}>
-        Select
-      </Button>
+      <Tooltip title={tooltipText ?? ""}>
+        <span>
```
Comment (vinceau) [inline src/renderer/components/PathInput.tsx:36]: Why do we need to wrap the `Button` in a span?

**Review by vinceau — CHANGES_REQUESTED** [review]

File: src/renderer/components/PathInput.tsx (hunk 6 above)
Comment (NikhilNarayana) [reply src/renderer/components/PathInput.tsx:36]: I think it could be a div or a span, but the reason is because if the Button is disabled the tooltip won't show

Comment (cnkeats) [reply src/renderer/components/PathInput.tsx:36]: This is correct. The tooltip will also be disabled if it's immediate child is disabled. Adding an empty wrapper element is the suggested solution.

https://mui.com/components/tooltips/

**Review by vinceau — APPROVED** [review]
LGTM
